springwolf-plugins/springwolf-cloud-stream-plugin/src/main/java/io/github/springwolf/plugins/cloudstream/asyncapi/scanners/common/FunctionalChannelBeanBuilder.java

```diff
@@ -25,19 +25,20 @@ public class FunctionalChannelBeanBuilder {
     public Set<FunctionalChannelBeanData> build(AnnotatedElement element) {
         Class<?> type = getRawType(element);
 
-        if (Consumer.class.isAssignableFrom(type)) {
-            Class<?> payloadType = getTypeGenerics(element).get(0);
+        List<Class<?>> typeGenerics = getTypeGenerics(element);
+        if (Consumer.class.isAssignableFrom(type) && typeGenerics.size() >= 1) {
+            Class<?> payloadType = typeGenerics.get(0);
             return Set.of(ofConsumer(element, payloadType));
         }
 
-        if (Supplier.class.isAssignableFrom(type)) {
-            Class<?> payloadType = getTypeGenerics(element).get(0);
+        if (Supplier.class.isAssignableFrom(type) && typeGenerics.size() >= 1) {
+            Class<?> payloadType = typeGenerics.get(0);
             return Set.of(ofSupplier(element, payloadType));
         }
 
-        if (Function.class.isAssignableFrom(type)) {
-            Class<?> inputType = getTypeGenerics(element).get(0);
-            Class<?> outputType = getTypeGenerics(element).get(1);
+        if (Function.class.isAssignableFrom(type) && typeGenerics.size() >= 2) {
```

Comment:
Might be too defensive, `Consumer` and `Supplier` always have exactly one generic type and `Function` always has exactly 2, but that wouldn't hurt.